src/transport.rs

```diff
@@ -1,6 +1,83 @@
-mod channel;
-pub mod client;
-mod handshake;
-pub mod io_stream;
-pub mod pem;
-pub mod server;
+use crate::customer;
+use crate::transport::client::Address;
+use http::uri::InvalidUri;
+use http::Uri;
+use std::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+use thiserror::Error;
+use webpki::{DnsName, DnsNameRef, InvalidDnsNameError};
+
+use transport::client;
```

Comment:
We have been using a more consolidated use structure in most of the crate, but it has to be done manually. Maybe we can find a way to explicitly fix this via `fmt`. These lines would be

```
use crate::{customer, transport::...};
use {
  http::{uri::InvalidUri, Uri},
  std::{fmt::{self, Display}, str::FromStr},
  thiserror::Error,
  ...
}
```

I think the benefits are consolidation (easier to scan), automatic alphabetization by fmt.

Reply:
I almost never look at the imports (except when I'm having some conflict), so I'm okay with whatever style we choose. However, if it doesn't happen automatically (preferably cargo fmt, maybe just IDE), I'm very much opposed to any choice of style, because that would mean manual work. There is this unstable rustfmt feature: https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#imports_granularity, I believe what you are suggesting is somewhat close to the "Crate" option, although it might be even closer to the "One" option. Unfortunately, the "One" option doesn't seem to work at this point. We can discuss what option seems most reasonable, or if there is another way to enforce it.

Reply:
Good feature. Per discussion offline, I'd be happy with `Crate` level formatting - I'm not particularly tied to what we currently have. The [tracking issue](https://github.com/rust-lang/rustfmt/issues/4991) implies that there are stability problems but doesn't cite any specific ones other than losing comments in the includes, which we don't have, and [there's an anecdote](https://github.com/LemmyNet/lemmy/blob/main/.rustfmt.toml) that `Crate`-level formatting has been stable for a while.

Since this repo already uses nightly rust, I think you can add a `.rustfmt.toml` file like the one linked above and specify this. I would expect that formatting config will automatically propagate to CI.